runtime/oti/VMHelpers.hpp

```diff
@@ -1373,8 +1373,7 @@ class VM_VMHelpers
 	static VMINLINE void
 	reportFinalFieldModified(J9VMThread* currentThread, J9Class* fieldClass)
 	{
-#if 0
-		if (J9_ARE_NO_BITS_SET(fieldClass->classFlags, J9ClassHasIllegalFinalFieldModifications)) {
+		if (J9_ARE_NO_BITS_SET(fieldClass->classFlags, J9ClassHasIllegalFinalFieldModifications) && (J9ClassInitSucceeded == fieldClass->initializeStatus)) {
```

Comment:
Please add a comment to describe why this is only reported after the class is initialized.  It makes it obvious that to the next person reading / debugging this.

Reply:
added